apps/www/server/index.js

```diff
@@ -18,8 +16,18 @@ const PORT = process.env.PORT || 3005
 
 const { CURTAIN_MESSAGE } = process.env
 
+if (!process.env.PUBLIC_BASE_URL) {
+  throw new Error(
+    'missing PUBLIC_BASE_URL environment variable, but is required by next-js middleware.',
+  )
+}
+
 const app = next({
   dev: DEV,
+  port: PORT,
+  hostname: process.env.PUBLIC_BASE_URL
+    ? new URL(process.env.PUBLIC_BASE_URL).hostname
+    : undefined,
```

Comment:
Since we now never get here without a PUBLIC_BASE_URL we can just so this:
```suggestion
  hostname: new URL(PUBLIC_BASE_URL).hostname
```